lib/src/main/scala/spinal/lib/bus/amba4/axi/sim/Axi4Master.scala

```diff
@@ -249,7 +249,12 @@ case class Axi4Master(axi: Axi4, clockDomain: ClockDomain, name: String = "unnam
         case _ => data.take(bytes)
       }
       val remaining = data.drop(slice.length)
-      writeSingle(addr, slice, id, burst, len, size)(handleTransaction(addr, transactionId, remaining))
+      val thisLen = if (remaining.length == 0) {
+        ((slice.length / bytePerBeat.toFloat).ceil.toInt - 1) min len
+      } else {
+        len
+      }
+      writeSingle(addr, slice, id, burst, thisLen, size)(handleTransaction(addr, transactionId, remaining))
```

Comment:
Not sure if silently down-capping `AxLEN` is a good idea, with the current semantics of `len`.  I see one of two ways: 
- change `len` to something like `maxLen`, so that it's used to limit the max burst size, or
- assert false here to complain that the user specified too big of `AxLEN`

Reply:
changed the parameter name to maxLen with a comment in the scaladoc.

I really don't like to assert here because this would require the user to specify a len that is compatible with the length of the data array and size.